packages/sdk/src/models/index.ts

```diff
@@ -222,15 +224,28 @@ export default class Models {
         const unsub = await tx.signAndSend(
           signer.address,
           { signer: signer.signer },
-          (result) =>
+          (result) => {
+            setTimeout(() => {
+              if (result.dispatchError || result.internalError) {
+                this.ipfsClient.unpinCidFromCluster(cid.toString());
+              }
+            });
```

Comment:
using set timeout to prevent the call from block the callback that has higher priority.

Reply:
Should we log the result of the unpinning? Would be beneficial in catching errors if any